### ISSUE TITLE
ProposalCreated event never emits signatures

### ISSUE THREAD

**Issue by GeraldHost** [OpenZeppelin/openzeppelin-contracts#3126]
**💻 Environment**

`4.4.1`

**📝 Details**

It seems that the ProposalCreated event will never emit signatures. I'm assuming that the intention was that this event should support signatures as to be use with the GovBravo compatibility module because `Governor.sol` doesn't support a propose method with signature strings.

**🔢 Code to reproduce bug**

The event interface is:
```solidity
event ProposalCreated(
    uint256 proposalId,
    address proposer,
    address[] targets,
    uint256[] values,
    string[] signatures,
    bytes[] calldatas,
    uint256 startBlock,
    uint256 endBlock,
    string description
);
```

When the event is emitted from `Governor.propose` the signatures value is set to `new string[](targets.length)`. Which if I understand correctly will always just be an array of empty strings of length `targets.length`. 

The GovernorBravo compatibility module then extends propose with this implementation.

```solidity
function propose(
    address[] memory targets,
    uint256[] memory values,
    string[] memory signatures,
    bytes[] memory calldatas,
    string memory description
) public virtual override returns (uint256) {
    _storeProposal(_msgSender(), targets, values, signatures, calldatas, description);
    return propose(targets, values, _encodeCalldata(signatures, calldatas), description);
}
```
Because the signatures are being encoded before getting passed `Governor.propose` the signatures will only ever be empty. 

I understand that I can get the signatures from the `_proposalDetails` mapping. It therefore seems like signatures can be removed from the event as it will never be populated? I can't see anywhere in the docs that explains why it should be there but perhaps there is a legit reason.

**Comment by Amxx** [comment]
Hello @GeraldHost 

As you pointed out, our intention was to keep `GovernorBravo` compatibility, particularly for indexing services that watch events. For this reason, we are keeping the same event signature, with the same parameter, and leaving that empty. 

Note that a `ProposalCreated` with empty "signatures" is valid by `GovernorBravo`'s standard (signatures can be left empty which result in a similar behavior to our contracts).

TLDR: even if we never populate it, we will keep this field for bravo compatibility.

**Comment by Amxx** [comment]
The bravo compatibility is here to "complete" bravo compatibility. But even without this this module you are already getting a lot for bravo compatibility, events being the most important point (also, using block number instead of timestamp, same status enum, ...)
